r/sedonadb: Cannot install via `pak::pkg_install()` due to `-Werror` flag on compiling aws-lc-sys

In https://github.com/apache/sedona-db/pull/146, I thought I succeeded to enable `pak::pkg_install()`, but I keep getting this error (to be clear, now I'm on macOS and away from Windows Desktop I was using during #146). It's weird that this doesn't happen when I compile the project by `devtools::load_all()` locally; it seems `-Werror` flag is passed from somewhere (not the `CFLAGS` in `Makevars`). Does anyone see the same error? Or, is something wrong with my setup?


```r
pak::pkg_install("apache/sedona-db/r/sedonadb")
```
```
  --- stderr

  ..snip..

  /Users/yutani/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/aws-lc-sys-0.32.0/aws-lc/crypto/fipsmodule/hmac/hmac.c:139:1: error: extra ';' outside of a function [-Werror,-Wextra-semi]
    139 | MD_TRAMPOLINES_EXPLICIT_PRECOMPUTED(MD5, MD5_CTX, MD5_CBLOCK)
        | ^
  /Users/yutani/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/aws-lc-sys-0.32.0/aws-lc/crypto/fipsmodule/hmac/hmac.c:120:60: note: expanded from macro 'MD_TRAMPOLINES_EXPLICIT_PRECOMPUTED'
    120 |   MD_TRAMPOLINES_EXPLICIT(HASH_NAME, HASH_CTX, HASH_CBLOCK);                  \
        |                                                            ^
  /Users/yutani/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/aws-lc-sys-0.32.0/aws-lc/crypto/fipsmodule/hmac/hmac.c:140:1: error: extra ';' outside of a function [-Werror,-Wextra-semi]
    140 | MD_TRAMPOLINES_EXPLICIT_PRECOMPUTED(SHA1, SHA_CTX, SHA_CBLOCK)
        | ^
  /Users/yutani/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/aws-lc-sys-0.32.0/aws-lc/crypto/fipsmodule/hmac/hmac.c:120:60: note: expanded from macro 'MD_TRAMPOLINES_EXPLICIT_PRECOMPUTED'
    120 |   MD_TRAMPOLINES_EXPLICIT(HASH_NAME, HASH_CTX, HASH_CBLOCK);                  \
        |                                                            ^

  ..snip..

  8 errors generated.
  make[3]: *** [aws-lc/crypto/fipsmodule/CMakeFiles/fipsmodule.dir/bcm.c.o] Error 1
  make[3]: *** Waiting for unfinished jobs....
  make[2]: *** [aws-lc/crypto/fipsmodule/CMakeFiles/fipsmodule.dir/all] Error 2
  make[2]: *** Waiting for unfinished jobs....
  make[1]: *** [all] Error 2
```

--- a/r/sedonadb/src/Makevars.in
+++ b/r/sedonadb/src/Makevars.in
@@ -30,6 +30,11 @@ PKG_LIBS = -L$(LIBDIR) -lsedonadbr @PKG_LIBS@
 
 CARGO_BUILD_ARGS = --lib --profile $(PROFILE) $(FEATURE_FLAGS) --manifest-path=./rust/Cargo.toml --target-dir $(TARGET_DIR)
 
+# "-Wno-pedantic" is to override the "-pedantic" flag passed by pkgbuild, which
+# makes the compilation of aws-lc-sys fail. Hopefully, such C code will be fixed
+# on upstream so that we can remove this.
+CFLAGS_TWEAKED = $(CFLAGS) -Wno-pedantic
+
 all: $(SHLIB) clean_intermediate
 
 $(SHLIB): $(STATLIB)
@@ -40,7 +45,7 @@ $(STATLIB):
 	# therefore is only used if cargo is absent from the user's PATH.
 	export PATH="$(PATH):$(HOME)/.cargo/bin" && \
 	  export CC="$(CC)" && \
-	  export CFLAGS="$(CFLAGS)" && \
+	  export CFLAGS="$(CFLAGS_TWEAKED)" && \
 	  export AWS_LC_SYS_CMAKE_BUILDER=1 && \
 	  export RUSTFLAGS="$(RUSTFLAGS)" && \
 	  if [ "$(TARGET)" != "wasm32-unknown-emscripten" ]; then \

--- a/r/sedonadb/src/Makevars.win.in
+++ b/r/sedonadb/src/Makevars.win.in
@@ -37,9 +37,10 @@ CARGO_LINKER = x86_64-w64-mingw32.static.posix-gcc.exe
 # cf. https://stackoverflow.com/a/68979087
 PKG_CONFIG_SYSROOT_DIR = /x86_64-w64-mingw32.static.posix/
 
-# TODO: should $(CFLAGS) be included here?
-# Do not pass -pedantic
-CFLAGS_TWEAKED = $(subst -pedantic,,$(ALL_CPPFLAGS))
+# "-Wno-pedantic" is to override the "-pedantic" flag passed by pkgbuild, which
+# makes the compilation of aws-lc-sys fail. Hopefully, such C code will be fixed
+# on upstream so that we can remove this.
+CFLAGS_TWEAKED = $(CFLAGS) -Wno-pedantic
 
 all: $(SHLIB) clean_intermediate
 

--- a/r/sedonadb/src/Makevars.win.in
+++ b/r/sedonadb/src/Makevars.win.in
@@ -52,11 +52,6 @@ $(STATLIB):
 	# in actual, but we need this tweak to please the compiler.
 	mkdir -p $(LIBDIR)/libgcc_mock && touch $(LIBDIR)/libgcc_mock/libgcc_eh.a
 
-	# AWS_LC_SYS_CFLAGS are not ideal: It would be better to use the
-	# CMake builder but this incurs path length errors on at least one Windows builder.
-	# The AWS_LC_SYS_CMAKE_BUILDER environment variable can still be set at a
-	# higher level on Windows environments that support it.
-
 	export CARGO_TARGET_X86_64_PC_WINDOWS_GNU_LINKER="$(CARGO_LINKER)" && \
 	  export LIBRARY_PATH="$${LIBRARY_PATH};$(LIBDIR)/libgcc_mock" && \
 	  export PKG_CONFIG_SYSROOT_DIR="$(PKG_CONFIG_SYSROOT_DIR)" && \

